### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Audio/Track/TrackManager.cs [hunk 1]
```diff
@@ -9,6 +11,8 @@ public class TrackManager : AudioCollectionManager<AudioTrack>
     {
         IResourceStore<byte[]> store;
 
+        AudioTrack exclusiveTrack;
```
Comment (Tom94) [inline osu.Framework/Audio/Track/TrackManager.cs]: Can you add a comment describing what an exclusive track is and what its purpose (expected behavior) is?

File: osu.Framework/Graphics/Transformations/Transform.cs [hunk 2]
```diff
@@ -23,31 +23,17 @@ public abstract class Transform<T> : ITransform
...
-
-        public bool IsAlive
+        public bool IsAliveAt(double time)
         {
-            get
-            {
-                //we could not be completely initialised yet.
-                if (Clock == null)
-                    return true;
+            CurrentTime = time;
```
Comment (Tom94) [inline osu.Framework/Graphics/Transformations/Transform.cs]: I don't like that this method does state changes. Its name does not indicate it should, so I'd be in favor of renaming it in a way that makes clear we are editing state. (Or setting the current time in a more explicit fashion somewhere else.)

File: osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs [hunk 3]
```diff
@@ -33,10 +33,6 @@ public override bool Initialize(BasicGameHost host)
             {
                 state = OpenTK.Input.Mouse.GetCursorState();
                 nativePosition = host.Window.PointToClient(new Point(state.X, state.Y));
-
-                //we want to still receive position updated when inactive, so reset the state only after reading the position.
-                if (!host.IsActive)
```
Comment (Tom94) [inline osu.Framework.Desktop/Input/Handlers/Mouse/OpenTKMouseHandler.cs]: I'd prefer this over all the `host.IsActive` checks below. Could you elaborate on why this change fixes mouse wheel?

File: osu.Framework/Audio/Track/TrackManager.cs [hunk 4]
```diff
@@ -30,10 +30,11 @@ public void SetExclusive(AudioTrack track)
             if (exclusiveTrack == track) return;
 
             Items.ForEach(i => i.Stop());
-            
-            AddItem(track);
 
+            exclusiveTrack?.Dispose();
```
Comment (peppy) [inline osu.Framework/Audio/Track/TrackManager.cs]: for now yes. as i mentioned, not sure how long this logic will remain; it's just what works for now.

items will get removed after they are disposed (fulfills the HasCompleted condition).
